lib/src/Uncertainty/Distribution/CombinationsDistribution.cxx

```diff
@@ -0,0 +1,518 @@
+    for (UnsignedInteger i = 0; i < j; ++i)
+      spearman(i, j) = getMarginal({i, j}).getSpearmanCorrelation()(0, 1);
+  return spearman;
+}
+
+/* Compute the Kendall concordance of the distribution */
+CorrelationMatrix CombinationsDistribution::getKendallTau() const
+{
+  CorrelationMatrix kendall(dimension_);
+  for (UnsignedInteger j = 0; j < dimension_; ++j)
+    for (UnsignedInteger i = 0; i < j; ++i)
+      kendall(i, j) = getMarginal({i, j}).getKendallTau()(0, 1);
+  return kendall;
+}
+
+/* Check if the distribution is continuous */
+Bool CombinationsDistribution::isContinuous() const
+{
+  return false;
+}
+
+/* Check if the distribution is discrete */
+Bool CombinationsDistribution::isDiscrete() const
+{
+  return true;
+}
+
+/* Tell if the distribution is integer valued */
+Bool CombinationsDistribution::isIntegral() const
+{
+  return true;
+}
+
+/* Parameters value and description accessor */
+CombinationsDistribution::PointWithDescriptionCollection CombinationsDistribution::getParametersCollection() const
+{
+  const UnsignedInteger dimension = getDimension();
+  PointWithDescriptionCollection parameters((dimension == 1 ? 1 : dimension + 1));
+  for (UnsignedInteger i = 0; i < dimension; ++i)
+  {
+    PointWithDescription point({static_cast<Scalar>(n_)});
+    point.setDescription({"n"});
+    point.setName(getDescription()[i]);
+    parameters[i] = point;
+  }
+  if (dimension > 1)
+  {
+    PointWithDescription point(getParameter());
+    point.setDescription(getParameterDescription());
+    point.setName("dependence");
+    parameters[dimension] = point;
+  }
+  return parameters;
+}
+
+/* Parameters value accessor */
+Point CombinationsDistribution::getParameter() const
+{
+  return {static_cast<Scalar>(k_), static_cast<Scalar>(n_)};
+}
+
+void CombinationsDistribution::setParameter(const Point & parameter)
+{
+  if (parameter.getSize() != 2) throw InvalidArgumentException(HERE) << "CombinationsDistribution expected 2 parameters, got " << parameter.getSize();
+  if (parameter[0] != std::round(parameter[0])) throw InvalidArgumentException(HERE) << "CombinationsDistribution first parameter k must be an integer, got " << parameter[0];
+  if (parameter[1] != std::round(parameter[1])) throw InvalidArgumentException(HERE) << "CombinationsDistribution second parameter n must be an integer, got " << parameter[1];
+  const Scalar w = getWeight();
+  *this = CombinationsDistribution(parameter[0], parameter[1]);
+  setWeight(w);
+}
+
+/* Parameters description accessor */
+Description CombinationsDistribution::getParameterDescription() const
+{
+  return {"k", "n"};
+}
+
+/* K accessor */
+void CombinationsDistribution::setK(const UnsignedInteger k)
+{
+  if (k == 0) throw InvalidArgumentException(HERE) << "Error: k must be > 0.";
+  if (k > n_) throw InvalidArgumentException(HERE) << "Error: k must be less or equal to n, here k=" << k << " and n=" << n_;
+  if (k != k_)
+  {
+    k_ = k;
```

Comment:
I cannot see why from the code. Isn'it consistent with the definition of the distribution:

![image](https://github.com/user-attachments/assets/efac8443-aabb-4a65-ab64-c1cca7b44b23)

This implies:

$$
\log\left(\mathbb{P}(\boldsymbol{X} = \boldsymbol{x})\right) = \log(1) - \log\left(\binom{n}{k}\right)
$$

and since $\log(1) = 0$, then:

$$
\log\left(\mathbb{P}(\boldsymbol{X} = \boldsymbol{x})\right) = - \log\left(\binom{n}{k}\right).
$$

This is why the right hand side must be `-LogBinomialCoefficient(n,k)`, isn't it?

Reply:
the difference comes from the fact that combinations are ordered